modules/mass.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * This module adds MASS support to Prebid.js.
+ */
+
+import { config } from '../src/config.js';
+import { getHook } from '../src/hook.js';
+import find from 'core-js-pure/features/array/find.js';
+
+let listenerAdded = false;
+let massEnabled = false;
+
+const defaultCfg = {
+  bootloaderUrl: 'https://cdn.massplatform.net/bootloader.js',
+  dealIdPattern: /^MASS/i
+};
+let cfg;
+
+const massBids = {};
+
+init();
+config.getConfig('mass', config => init(config.mass));
+
+/**
+ * Module init.
+ */
+function init(customCfg) {
+  cfg = Object.assign({}, defaultCfg, customCfg);
+
+  if (cfg.enabled === false) {
+    if (massEnabled) {
+      massEnabled = false;
+      getHook('addBidResponse').getHooks({hook: addBidResponseHook}).remove();
+    }
+  } else {
+    if (!massEnabled) {
+      getHook('addBidResponse').before(addBidResponseHook);
+      massEnabled = true;
+    }
+  }
+}
+
+/**
+ * Before hook for 'addBidResponse'.
+ */
+export function addBidResponseHook(next, adUnitCode, bid) {
+  if (!isMassBid(bid)) {
+    return next(adUnitCode, bid);
+  }
+
+  const bidRequest = find(this.bidderRequest.bids, bidRequest =>
+    bidRequest.bidId === bid.requestId
+  );
+
+  massBids[bid.requestId] = {
+    bidRequest,
+    bid,
+    adm: bid.ad
+  };
+
+  bid.ad = '<script>window.top.postMessage({massBidId: "' + bid.requestId + '"}, "*");\x3c/script>';
```

Comment:
Maybe here you need `window.parent` and not `window.top`??  
Will this work if pbjs is running inside iframe and not on `window.top`?

Reply:
yes, window.parent makes more sense, updated, thanks!